src/hotspot/share/utilities/tribool.hpp

```diff
@@ -73,7 +77,8 @@ class TriBoolArray {
 
     TriBoolAssigner& operator=(bool newval) {
       _slot ^= ((u1)_value) << _offset;  // reset the tribool
-      _value = (u1)newval| 2;
+      _value = newval;
+      _value |= 2; // set to not-default
```

Comment:
```suggestion
      TriBool::operator=(newval);
```

Reply:
I don't think this is more readable.  I believe you that it doesn't give -Wconversion warnings though.